src/aposteriori/data_prep/create_frame_data_set.py

```diff
@@ -120,7 +245,14 @@ def create_residue_frame(
         The number of voxels per edge that the cube of space will be converted into i.e.
         the final cube will be `voxels_per_side`^3. This must be a odd, positive integer
         so that the CA atom can be placed at the centre of the frame.
-    
+    encode_cb: bool
+        Whether to encode the Cb at an average position in the frame.
+    atom_encoder: str
+        Whether to encode the frames as C, N, O or have additional channels for Ca and Cb.
+        Options available are: "CNO", "CNOCB", "CNOCBCA".
+    encoder_length: int
+        Length of the number of channels needed in the encoded atoms.
```

Comment:
Wouldn't a simple dictionary be simpler? or do you also want it to encode the residues and the frames?

Reply:
A dict would be simpler, but a `dict` would require the user to define the encoding, rather than just supplying the labels. This would give you control over how the encoding is generated, and keeps all the code for creating the encoding encapsulated in the class.

Reply:
The thing is, from a user perspective, you can't really write a list of labels from the CLI. So I think the input from the cli should still be `"CNO", "CNOCB", "CNOCBCA".`

Perhaps it could be initialised with with the `atom_encoder` and a dictionary of labels will create the list of labels:

```
{
"CNO": [ "C","N", "O"],
"CNOCB": [ "C","N", "O", "CB"]
 }
```

This could essentially get rid of the encoder_length parameter. 

However, we'll still have the same problem for which this is needed:

```
    elif atom_label == "CA":
        if encoder_length == 3 or 4:
            # Add General C encoding
            atom_encoding[0] = True
        elif encoder_length == 5:
            atom_encoding[4] = True
```

Because in CNO and CNOCB, the CA encoding acts as a C, while in CNOCBCA it acts as a different atom.

Reply:
OOPS, forget it, I got it!